### ISSUE TITLE
Fix the problem with conditional input and CFG

### ISSUE THREAD

**Issue by chrisway613** [haidog-yaqub/MeanFlow#9]
Thanks for your great work!
In the `MeanFlow.loss` method, there may go something wrong when the condition input is `None`(i.e. unconditional case):

       if self.w is not None:
            uncond = torch.ones_like(c) * self.num_classes
            with torch.no_grad():
                u_t = model(z, t, t, uncond)
            v_hat = self.w * v + (1 - self.w) * u_t
        else:
            v_hat = v

        cfg_mask = torch.rand_like(c.float()) < self.cfg_ratio
        c = torch.where(cfg_mask, uncond, c)

        if self.cfg_uncond == 'v':
            # as v = wv - (1-w)v = wv - (1-w)u in the unconditional case, should we directly use v instead?
            cfg_mask = rearrange(r, "b -> b 1 1 1").bool()
            v_hat = torch.where(cfg_mask, v, v_hat)

like `torch.ones_like(c)` will occur error, and thus I change this part to below:

        if c is not None:
            uncond = torch.ones_like(c) * self.num_classes
            cfg_mask = torch.rand_like(c.float()) < self.cfg_ratio
            c = torch.where(cfg_mask, uncond, c)

            if self.cfg_uncond == 'u' and 0 < self.w < 1:
                with torch.no_grad():
                    u_t = model(z, t, t, uncond)
                # Equation (19) in the paper
                v_hat = self.w * v + (1 - self.w) * u_t
            else:
                # as v = wv + (1-w)v = wv + (1-w)u in the unconditional case, should we directly use v instead?
                v_hat = v
        else:
            v_hat = v

What about your opinion, or may I create a PR?

**Comment by haidog-yaqub** [comment]
> Thanks for your great work! In the `MeanFlow.loss` method, there may go something wrong when the condition input is `None`(i.e. unconditional case):
> 
> ```
>    if self.w is not None:
>         uncond = torch.ones_like(c) * self.num_classes
>         with torch.no_grad():
>             u_t = model(z, t, t, uncond)
>         v_hat = self.w * v + (1 - self.w) * u_t
>     else:
>         v_hat = v
> 
>     cfg_mask = torch.rand_like(c.float()) < self.cfg_ratio
>     c = torch.where(cfg_mask, uncond, c)
> 
>     if self.cfg_uncond == 'v':
>         # as v = wv - (1-w)v = wv - (1-w)u in the unconditional case, should we directly use v instead?
>         cfg_mask = rearrange(r, "b -> b 1 1 1").bool()
>         v_hat = torch.where(cfg_mask, v, v_hat)
> ```
> 
> like `torch.ones_like(c)` will occur error, and thus I change this part to below:
> 
> ```
>     if c is not None:
>         uncond = torch.ones_like(c) * self.num_classes
>         cfg_mask = torch.rand_like(c.float()) < self.cfg_ratio
>         c = torch.where(cfg_mask, uncond, c)
> 
>         if self.cfg_uncond == 'u' and 0 < self.w < 1:
>             with torch.no_grad():
>                 u_t = model(z, t, t, uncond)
>             # Equation (19) in the paper
>             v_hat = self.w * v + (1 - self.w) * u_t
>         else:
>             # as v = wv + (1-w)v = wv + (1-w)u in the unconditional case, should we directly use v instead?
>             v_hat = v
>     else:
>         v_hat = v
> ```
> 
> What about your opinion, or may I create a PR?

Thank you for pointing out! Please feel free to submit a PR.
